odoo/addons/base/models/ir_attachment.py

```diff
@@ -374,25 +371,17 @@ def check(self, mode, values=None):
             # ignore attachments that are not attached to a resource anymore
             # when checking access rights (resource was deleted but attachment
             # was not)
-            if res_model not in self.env:
-                require_employee = True
-                continue
-            elif res_model == 'res.users' and len(res_ids) == 1 and self._uid == list(res_ids)[0]:
+            if res_model == 'res.users' and len(res_ids) == 1 and self._uid == list(res_ids)[0]:
```

Comment:
I think we should keep
```
if res_model not in self.env:
    continue
```
Otherwise the below line records = self.env[res_model].browse(res_ids).exists() will crash.